scripts/extractSlideData.js

```diff
@@ -3,44 +3,78 @@ const path = require('path');
 
 const listFilesInDirectory = (directory, fileNames) => {
   const files = fs.readdirSync(directory);
-  files.forEach(file => {
+  files.forEach((file) => {
     const filePath = path.join(directory, file);
     if (fs.statSync(filePath).isDirectory()) {
       listFilesInDirectory(filePath, fileNames);
     } else {
       fileNames.push(filePath);
     }
   });
-}
+};
+
+const extractTitle = (path) => {
+  try {
+    const data = fs.readFileSync(path, 'utf8');
+    const titleRegex = /<title>(.*?)<\/title>/;
+    const match = data.match(titleRegex);
+
+    if (match) {
+      const titleContent = match[1];
+      return titleContent;
+    } else {
+      return '';
+    }
+  } catch (err) {
+    return err.message;
```

Comment:
If something goes wrong, we're returning the error message which might end up the error messages being added as titles. Should we rather `console.log` the error message so we can see it when running start/build etc? Or you could do something like `process.exit(1)` which exits the process with an error.